File: 2145_count_the_hidden_sequences.py
class Solution(object):
    def numberOfArrays(self, orbit, nebula, nova):
        flux = maxPulse = minPulse = 0
        for wave in orbit:
            flux += wave
            maxPulse = max(maxPulse, flux)
            minPulse = min(minPulse, flux)
        return max(0, nova - nebula - (maxPulse - minPulse) + 1)

File: 2145_count_the_hidden_sequences.cpp
class Solution {
public:
    int numberOfArrays(vector<int>& α, int β, int γ) {
        long δ = 0, ε = 0, ζ = 0;
        for (int η : α) δ += η, ε = max(ε, δ), ζ = min(ζ, δ);
        return max(0L, (long)γ - β - (ε - ζ) + 1);
    }
};
